optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -240,4 +245,87 @@ public void skipAhead() throws ExecutionException, InterruptedException {
+        solverManager.terminateEarly(secondProblemId);
+        waitForJobStatusChangeOrTimeout(solverManager, secondProblemId, SolverStatus.NOT_SOLVING);
+
+        // Reschedule second job. It reintroduces it to the solverManager by changing status from NOT_SOLVING to
+        // SOLVING_SCHEDULED in which it stays.
+        solverManager.solve(secondProblemId, PlannerTestUtils.generateTestdataSolution("s2"));
+        waitForJobStatusChangeOrTimeout(solverManager, secondProblemId, SolverStatus.SOLVING_SCHEDULED);
+
+        // Terminate the first job which triggers it to change it's status from SOLVING_ACTIVE to NOT_SOLVING.
+        // Automatically starts working on the second job. Its status changes from STATUS_SCHEDULED to STATUS_ACTIVE.
+        solverManager.terminateEarly(firstProblemId);
+        waitForJobStatusChangeOrTimeout(solverManager, firstProblemId, SolverStatus.NOT_SOLVING);
+        waitForJobStatusChangeOrTimeout(solverManager, secondProblemId, SolverStatus.SOLVING_ACTIVE);
+
+        // Terminate the second job, its status changes from SOLVING_ACTIVE to NOT_SOLVING. Both jobs stopped solving.
+        solverManager.terminateEarly(secondProblemId);
+        waitForJobStatusChangeOrTimeout(solverManager, secondProblemId, SolverStatus.NOT_SOLVING);
+        assertSame(SolverStatus.NOT_SOLVING, solverManager.getSolverStatus(firstProblemId));
+
+        // Reintroduce the problems to solverManager and attempt to close them all at once using solverManager.close().
+        // The first job's status changes from SOLVING_ACTIVE to NOT_SOLVING.
+        // The second job's status changes from SOLVING_SCHEDULED to NOT_SOLVING.
+        solverManager.solve(firstProblemId, PlannerTestUtils.generateTestdataSolution("s1"));
+        waitForJobStatusChangeOrTimeout(solverManager, firstProblemId, SolverStatus.SOLVING_ACTIVE);
+
+        solverManager.solve(secondProblemId, PlannerTestUtils.generateTestdataSolution("s2"));
+        assertSame(SolverStatus.SOLVING_SCHEDULED, solverManager.getSolverStatus(secondProblemId));
+
+        solverManager.close();
+        waitForJobStatusChangeOrTimeout(solverManager, firstProblemId, SolverStatus.NOT_SOLVING);
+        waitForJobStatusChangeOrTimeout(solverManager, secondProblemId, SolverStatus.NOT_SOLVING);
+    }
+
+    private void waitForJobStatusChangeOrTimeout(SolverManager<TestdataSolution, Long> solverManager, Long problemId, SolverStatus expectedStatusChange) throws InterruptedException {
+        long t = System.currentTimeMillis();
+        long end = t + 5000;
+        // Check every 5 milliseconds for solverJob status change for 5 seconds at maximum, otherwise time-out and fail.
+        while (t < end) {
+            if (solverManager.getSolverStatus(problemId) == expectedStatusChange) {
+                return;
+            }
+            TimeUnit.MILLISECONDS.sleep(5);
```

Comment:
Above the sleep, can you add these comments?

```
// Quick polling (not a Test Thread Sleep anti-pattern)
// Test is still fast on fast machines and doesn't randomly fail on slow machines.
```

Motivation: the use of Thread.sleep() is a red flag in tests, because it's often used in a catch 22 situation: wait long enough to succeed on the slow machine, but don't wait long so the tests run quickly. This comment makes it clear that this code is doing it correctly.